### PR TITLE
feature(collector): e2e tests for the collector

### PR DESCRIPTION
**Pull request by szdavid92**

### CODE REVIEW

File: e2e/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+var TRACE_SERVICE_NAME_TEST = 'service-name'
+
+var childProcessTest = require('./utils/childProcessTest')
+var test = require('tape')
+var waterfall = require('async').waterfall
+var path = require('path')
+
+test('Trace collector', function (t) {
+  waterfall([
+    childProcessTest.bind(
```
Comment (szdavid92) [inline e2e/index.js]: tried very hard to keep this code unclutter, this is the best i managed to achieve

File: e2e/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+var TRACE_API_KEY_TEST = 'api-key'
+var TRACE_SERVICE_NAME_TEST = 'service-name'
+
+var childProcessTest = require('./utils/childProcessTest')
+var test = require('tape')
```
Comment (peteyy) [inline e2e/index.js]: why do we need tape?

Comment (szdavid92) [inline e2e/index.js]: @peteyy because it generates tap output by default so it is very easy to report consistently in child processes

Comment (szdavid92) [inline e2e/index.js]: @peteyy or do you mean, why do we need it here, when we have no assertions? thats a good point

Comment (peteyy) [inline e2e/index.js]: What do you mean by tap output?

Comment (peteyy) [inline e2e/index.js]: @szdavid92 so does mocha 
```bash
mocha -R tap
```
but why do we need that?

**Pull request closed without merging**